File: src/main/java/com/example/stockfx/entities/Article.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.example.stockfx.entities;

/**
 *
 * @author nours
 */
public class Article {
    int ref_article,ref_magazine;
    float rate;
    String titre_article,contenu_article,auteur_article,genre_article,date_article;

    public Article( String titre_article, String contenu_article, String auteur_article, String genre_article, String date_article,int ref_magazine) {
        this.ref_magazine = ref_magazine;
        this.rate = rate;
        this.titre_article = titre_article;
        this.contenu_article = contenu_article;
        this.auteur_article = auteur_article;
        this.genre_article = genre_article;
        this.date_article = date_article;
    }

    

    public Article() {
        }

  
    public int getRef_article() {
        return ref_article;
    }

    public void setRef_article(int ref_article) {
        this.ref_article = ref_article;
    }

    public int getRef_magazine() {
        return ref_magazine;
    }

    public void setRef_magazine(int ref_magazine) {
        this.ref_magazine = ref_magazine;
    }

    public float getRate() {
        return rate;
    }

    public void setRate(float rate) {
        this.rate = rate;
    }

    public String getTitre_article() {
        return titre_article;
    }

    public void setTitre_article(String titre_article) {
        this.titre_article = titre_article;
    }

    public String getContenu_article() {
        return contenu_article;
    }

    public void setContenu_article(String contenu_article) {
        this.contenu_article = contenu_article;
    }

    public String getAuteur_article() {
        return auteur_article;
    }

    public void setAuteur_article(String auteur_article) {
        this.auteur_article = auteur_article;
    }

    public String getGenre_article() {
        return genre_article;
    }

    public void setGenre_article(String genre_article) {
        this.genre_article = genre_article;
    }

    public String getDate_article() {
        return date_article;
    }

    public void setDate_article(String date_article) {
        this.date_article = date_article;
    }

    @Override
    public String toString() {
        return "Article{" + "ref_article=" + ref_article + ", ref_magazine=" + ref_magazine + ", rate=" + rate + ", titre_article=" + titre_article + ", contenu_article=" + contenu_article + ", auteur_article=" + auteur_article + ", genre_article=" + genre_article + ", date_article=" + date_article + '}';
    }
    
}

File: src/Entity/Article.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Article
 *
 * @ORM\Table(name="article", indexes={@ORM\Index(name="fk_magazine", columns={"ref_magazine"})})
 * @ORM\Entity
 */
class Article
{
    /**
     * @var int
     *
     * @ORM\Column(name="ref_article", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $refArticle;

    /**
     * @var string
     *
     * @ORM\Column(name="titre_article", type="string", length=50, nullable=false)
     */
    private $titreArticle;

    /**
     * @var string
     *
     * @ORM\Column(name="contenu_article", type="text", length=65535, nullable=false)
     */
    private $contenuArticle;

    /**
     * @var string
     *
     * @ORM\Column(name="auteur_article", type="string", length=50, nullable=false)
     */
    private $auteurArticle;

    /**
     * @var string
     *
     * @ORM\Column(name="genre_article", type="string", length=50, nullable=false)
     */
    private $genreArticle;

    /**
     * @var string
     *
     * @ORM\Column(name="date_article", type="string", length=50, nullable=false)
     */
    private $dateArticle;

    /**
     * @var int
     *
     * @ORM\Column(name="ref_magazine", type="integer", nullable=false)
     */
    private $refMagazine;

    /**
     * @var int|null
     *
     * @ORM\Column(name="selected", type="integer", nullable=true)
     */
    private $selected = '0';

    /**
     * @var int
     *
     * @ORM\Column(name="vues", type="integer", nullable=false)
     */
    private $vues = '0';

    public function getRefArticle(): ?int
    {
        return $this->refArticle;
    }

    public function getTitreArticle(): ?string
    {
        return $this->titreArticle;
    }

    public function setTitreArticle(string $titreArticle): self
    {
        $this->titreArticle = $titreArticle;

        return $this;
    }

    public function getContenuArticle(): ?string
    {
        return $this->contenuArticle;
    }

    public function setContenuArticle(string $contenuArticle): self
    {
        $this->contenuArticle = $contenuArticle;

        return $this;
    }

    public function getAuteurArticle(): ?string
    {
        return $this->auteurArticle;
    }

    public function setAuteurArticle(string $auteurArticle): self
    {
        $this->auteurArticle = $auteurArticle;

        return $this;
    }

    public function getGenreArticle(): ?string
    {
        return $this->genreArticle;
    }

    public function setGenreArticle(string $genreArticle): self
    {
        $this->genreArticle = $genreArticle;

        return $this;
    }

    public function getDateArticle(): ?string
    {
        return $this->dateArticle;
    }

    public function setDateArticle(string $dateArticle): self
    {
        $this->dateArticle = $dateArticle;

        return $this;
    }

    public function getRefMagazine(): ?int
    {
        return $this->refMagazine;
    }

    public function setRefMagazine(int $refMagazine): self
    {
        $this->refMagazine = $refMagazine;

        return $this;
    }

    public function getSelected(): ?int
    {
        return $this->selected;
    }

    public function setSelected(?int $selected): self
    {
        $this->selected = $selected;

        return $this;
    }

    public function getVues(): ?int
    {
        return $this->vues;
    }

    public function setVues(int $vues): self
    {
        $this->vues = $vues;

        return $this;
    }


}
